power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -163,6 +166,9 @@ template <dataset_type_tag dataset_type_> class Dataset {
         if (!is_batch() && scenario > 0) {
             throw DatasetError{"Cannot export a single dataset with specified scenario\n"};
         }
+        if (scenario >= batch_size()) {
```

Comment:
I proposed to put an assert instead of exception. It is an UB if the user has an overflow, just as many places in the code.